File: tests/fixtures/discrimination/python/out_of_corpus/generic_12.py
class BankAccount:
    def __init__(self, owner, balance=0):
        self.owner = owner
        self.balance = balance

    def deposit(self, amount):
        self.balance += amount
        return self.balance

    def withdraw(self, amount):
        if amount > self.balance:
            return None
        self.balance -= amount
        return self.balance

    def get_balance(self):
        return self.balance

    def transfer(self, other, amount):
        if amount > self.balance:
            return False
        self.balance -= amount
        other.balance += amount
        return True

File: tests/fixtures/discrimination/rust/out_of_corpus/generic_12.rs
struct BankAccount {
    owner: String,
    balance: f64,
}

impl BankAccount {
    fn new(owner: &str, balance: f64) -> BankAccount {
        BankAccount {
            owner: String::from(owner),
            balance,
        }
    }

    fn deposit(&mut self, amount: f64) -> f64 {
        self.balance += amount;
        self.balance
    }

    fn withdraw(&mut self, amount: f64) -> Option<f64> {
        if amount > self.balance {
            None
        } else {
            self.balance -= amount;
            Some(self.balance)
        }
    }

    fn get_balance(&self) -> f64 {
        self.balance
    }

    fn get_owner(&self) -> &str {
        &self.owner
    }
}
